### DISCUSSION TITLE
Welcome to node_exporter_aix Discussions!

### DISCUSSION THREAD

**Discussion by thorhs** [thorhs/node_exporter_aix#4] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jgrig472** [comment]
Hi and thank you for putting this exporter together and making it available to download! I was wondering if you'd be willing to update the install line to be a little clearer. It'd be nice to know that to install the package you should specify the directory you have the **.bff** file in for the **-d** switch and to know that the package name will always be **node_exporter_aix** when going to install it. Initially I kept trying to use the full **.bff** file name when trying to install it. I ended up spending quite a bit of time trying to figure all this out initially as I hadn't used AIX until recently.
**sudo installp -ac -d /_(location of downloaded .bff file)_ node_exporter_aix**

**Reply by hugfdbc31** [reply]
lease provide bff file.. i am unable to generate  bff file in aix server.

i am strucked here , pleae help

**Comment by New77** [comment]
Hi, Could you please share the steps to compile.  I'm having error while I compile

**Reply by thorhs** [reply]
Hi, you need to use gmake (GNU make) from the AIX Toolbox to compile.  If you still have errors, could you please send the output of the compilation?

The command I use to compile is:
`gmake clean all`

To generate the .bff file is:
`gmake bff`

Hope this helps.

**Reply by New77** [reply]
/tmp/node #make bff
        g++ -Wall -Werror -fmax-errors=5 -fconcepts -std=c++17 -pthread -lperfstat -I Simple-Web-Server -I asio/asio/include -DUSE_STANDALONE_ASIO -DASIO_STANDALONE -DASIO_HAS_PTHREADS -c -o build/server.o server.cpp
In file included from /usr/include/sys/thread.h:40,
                 from /usr/include/sys/ptrace.h:28,
                 from /usr/include/sys/proc.h:43,
                 from /usr/include/sys/pri.h:43,
                 from /usr/include/sys/sched.h:38,
                 from /usr/include/sched.h:51,
                 from /usr/include/pthread.h:63,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr-posix.h:35,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr-default.h:30,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr.h:148,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/ext/atomicity.h:35,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/basic_string.h:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/string:52,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/stdexcept:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/optional:38,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/node_handle.h:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/stl_tree.h:72,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/map:60,
                 from server.cpp:2:
/usr/include/sys/time.h:108:16: error: redefinition of 'struct sigset_t'
 typedef struct sigset_t {
                ^~~~~~~~
In file included from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include-fixed/time.h:196,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/ctime:42,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/chrono:41,
                 from server.cpp:1:
/opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include-fixed/sys/types.h:345:16: note: previous definition of 'struct sigset_t'
 typedef struct sigset_t {
                ^~~~~~~~
In file included from /usr/include/sys/thread.h:40,
                 from /usr/include/sys/ptrace.h:28,
                 from /usr/include/sys/proc.h:43,
                 from /usr/include/sys/pri.h:43,
                 from /usr/include/sys/sched.h:38,
                 from /usr/include/sched.h:51,
                 from /usr/include/pthread.h:63,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr-posix.h:35,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr-default.h:30,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/powerpc-ibm-aix7.2.0.0/pthread/bits/gthr.h:148,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/ext/atomicity.h:35,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/basic_string.h:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/string:52,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/stdexcept:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/optional:38,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/node_handle.h:39,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/bits/stl_tree.h:72,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/map:60,
                 from server.cpp:2:
/usr/include/sys/time.h:124:3: error: conflicting declaration 'typedef int sigset_t'
 } sigset_t;
   ^~~~~~~~
In file included from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include-fixed/time.h:196,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/ctime:42,
                 from /opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include/c++/chrono:41,
                 from server.cpp:1:
/opt/freeware/lib/gcc/powerpc-ibm-aix7.2.0.0/8.3.0/include-fixed/sys/types.h:361:3: note: previous declaration as 'typedef struct sigset_t sigset_t'
 } sigset_t;
   ^~~~~~~~
server.cpp:14:20: error: 'SimpleWeb' does not name a type; did you mean 'Simple_lock'?
 using HttpServer = SimpleWeb::Server<SimpleWeb::HTTP>;
                    ^~~~~~~~~
                    Simple_lock
server.cpp: In function 'int start_server(int, int)':
server.cpp:19:2: error: 'HttpServer' was not declared in this scope
  HttpServer server;
  ^~~~~~~~~~
server.cpp:19:2: note: suggested alternative: 'start_server'
  HttpServer server;
  ^~~~~~~~~~
  start_server
server.cpp:20:2: error: 'server' was not declared in this scope
  server.config.port = port;
  ^~~~~~
server.cpp:20:2: note: suggested alternative: 'strsep'
  server.config.port = port;
  ^~~~~~
  strsep
compilation terminated due to -fmax-errors=5.
make: The error code from the last command is 1.


Stop.

**Reply by thorhs** [reply]
This looks like an update of gcc is required: [See discussion on IBM forum](https://community.ibm.com/community/user/power/communities/community-home/digestviewer/viewthread?GroupId=6211&MessageKey=7af2f62f-f23f-40cb-aa87-048625dad735&CommunityKey=10c1d831-47ee-4d92-a138-b03f7896f7c9&ReturnUrl=%2Fcommunity%2Fuser%2Fpower%2Fcommunities%2Fcommunity-home%2Fdigestviewer%3FListKey%3Daf5415f8-d8d8-4ac4-b3cb-08a8da184054)

Could you see if an upgrade of gcc helps?

**Reply by New77** [reply]
I still see same error after upgraded to gcc8-8.3.0-6.

**Reply by thorhs** [reply]
Hmmmm, I haven't had the chance to update my dev server so I can't test this out locally just yet.  I'll get this on my radar to check soon, but there is a big project going on at the moment so I can't say when I can get to it.

Could you please create an issue regarding this, it has more visibility than a discussion.  Also, if you continue trying and get it to work to update the issue for others to see?

**Reply by hugfdbc31** [reply]
please provide bff file..  i am unable to generate in aix server.

i am strucked here , pleae help

**Comment by sumanbackup28** [comment]
Please share latest bff file or please share end to end process and repo to build bff.
